anomalib/models/ganomaly/torch_model.py

```diff
@@ -0,0 +1,227 @@
+"""Torch models defining encoder, decoder, Generator and Discriminator."""
```

Comment:
Reminder: We need to check if this code contains any duplication from DCGAN repo. If it does, we should either acknowledge the authors (depending on the license) or write our own implementation.